strawberryfields/program.py

```diff
@@ -546,7 +546,15 @@ def _linked_copy(self):
             Program: a copy of the Program
         """
         self.lock()
-        p = copy.copy(self)  # shares RegRefs with the source
+        p = copy.copy(self)
+
+        for name, val in self.__dict__.items():
+            # Deep-copy all attributes except 'circuit' and 'reg_refs', since the programs
+            # should share the same register references. Program.circuit potentially
+            # contains FreeParameters/MeasuredParameters, which contain RegRefs.
+            if name not in ("circuit", "reg_refs", "init_reg_refs"):
+                setattr(p, name, copy.deepcopy(val))
```

Comment:
Optimally, we should deep-copy the circuit as well, but since the may contain regrefs, this isn't as straight-forward to do. Just making a deepcopy here would cause two errors:
1. Due to `MeasuredParameter` not supporting `deepcopy` (because of the different signature of the `__new__()` method). A `__deepcopy__()` method would potentially need to be added.
2. Any copied regrefs would raise the unnerving `"RegRef state has become inconsistent."` issue we had [earlier](https://github.com/XanaduAI/strawberryfields/issues/683).

We could e.g., override the `__deepcopy__` method for the `Command` class, but that would require it to only deep-copy everything _except_ symbolic parameters. Alternatively, add a `copy_everything_except_regref` method to `Command`.

Reply:
Because this is a potentially actionable item, maybe is worth to have it as a (TODO) comment in the code. It might get lost here.

Reply:
Because this is a potentially actionable item, maybe is worth to have it as a (TODO) comment in the code. It might get lost here.

Reply:
I've created an issue for this instead (#691). I think it's better to keep track of it there rather than as a TODO which can easily be lost. 🙂

Reply:
I've created an issue for this instead (#691). I think it's better to keep track of it there rather than as a TODO which can easily be lost. 🙂